### DISCUSSION TITLE
Postgres Connection Failure : pga_hba.conf entry for host .. , SSL off

### DISCUSSION THREAD

**Discussion by o1lab** [nocodb/nocodb#323] [Q&A] [answered]
While connecting to postgres database (specially in DigitalOcean) 

There are errors seen like below

<img width="816" alt="Screen Shot 2021-07-05 at 17 28 37" src="https://user-images.githubusercontent.com/5435402/124468431-b7100980-ddb6-11eb-813f-dce3c372fea8.png">

**Comment by o1lab** [comment, marked as answer]
NocoDB uses knexjs for connecting to databases. And we allow connection params to be changed under the hood like below 

https://www.youtube.com/watch?v=rS0NRwRCjx0&

So you can set SSL to true like below

  "connection": {
    "host": "localhost",
    "port": "5432",
    "user": "postgres",
    "password": "password",
    "database": "postgres",
    **"ssl": "true"**
  },

and test database connection again.

This should help!

**Reply by gabrielalmeida** [reply]
I'm seeing the same while trying to connect to a Heroku Postgres database. The `ssl` prop is always overwritten to empty after clicking `Test Database Connection` and the same error is shown.

Even if I cheat and try to bypass it by adding `?ssl=true` to the `database` field (assuming it would append it to the underlying knex connectionString's queryString), it will result in the same error, never allowing knex to connect with SSL set to `true`.

**Reply by o1lab** [reply]
Did you try 1-Click heroku from our website ?

**Reply by gabrielalmeida** [reply]
Yes, this setup was done using the 1-Click installer and the "local" database that's attached by default works as expected, though, I'm willing to connect it to an existent external database, which is this other Postgres instance that happens to also be hosted on Heroku.

**Reply by jsanta** [reply]
Hi @gabrielalmeida 
Faced some similar issue and had to dig a little on the Knex JS connection parameters. You'l have to edit the connection JSON when creating a project that connects to a external database :
```
  "client": "pg",
  "connection": {
    "host": "your_heroku_host",
    "port": "5432",
    "user": "your_heroku_db_user",
    "password": "your_heroku_db_pasword",
    "database": "your_heroku_db",
    "ssl": {
      "require": true,
      "rejectUnauthorized": false,
      "sslMode": "no-verify"
    }
  },
  "searchPath": [ "the_schema_you're_wiling_to_use" ],
```
I have my doubts on the sslMode parameter, but at least it doesn't  explode.
We also had to add the searchPath as we are using a specific schema for some connections.

Hope it helps. Best regards from Chile and stay safe,
José Ignacio Santa Cruz

**Reply by o1lab** [reply]
Excellent, thank you for the notes

**Reply by rambhual** [reply]
Thanks you @jsanta
